README.md

```diff
@@ -21,3 +21,7 @@ To run your tests
 ### 401 Data Structures, Code Challenges
 
 - Please follow the instructions specific to your 401 language, which can be found in the directory below, matching your course.
+
+## Table of Contents
+
+[Array Reverse](./javascript/code-challenges/array-reverse/README.md)
```

Comment:
Where you have it now is fine, you can also have a separate folder for all 301 code challenges and 401 challenges if you want. During the first week no actual code is needed except for the whiteboards!